lib/3scale_toolbox/commands/plans_command/import/step.rb

```diff
@@ -82,8 +98,14 @@ def find_feature_by_system_name(system_name)
             service_features.find { |feature| feature['system_name'] == system_name }
           end
 
-          def find_metric_by_system_name(system_name)
-            service_metrics_and_methods.find { |metric| metric.system_name == system_name }
+          def find_metric(metric_system_name, backend_system_name)
```

Comment:
The name should probably be `metric_backend_system_name`

Reply:
the callers of this ruby method have
* metric system_name
* (optionally) backend system name

They want to get the actual metric object out of it. Also, this method is usefull to check if the metric reference is valid (the metric exists or not)

It is not always a backend's metric. That's why I think that is not a good name. But I am not good naming methods. Propose an alternative :)

Reply:
I am still not fully sure I understand the purpose of the method.
`They want to get the actual metric object out of it`: Is the purpose then to find a product metric, or a backend metric depending on the provided parameters? If that's the case, why not having a `find_backend_metric` and a `find_product_metric`?

Reply:
it would be clearer if I implement  
```ruby
class Backend
  def find_metric_or_method(system_name)
  end
end

class Service
  def find_metric_or_method(system_name)
  end
end
```

?? 

Then from import plan command, when importing limits and pricing rules, we get service or backend (depending on backend system name exists or not) and then call `find_metric_or_method` method to get the metric (or method).

it would be something like this
```ruby
metric_owner = if pricing_rule['backend_system_name'].nil?
                 service
               else
                  Entities::Backend.find_by_system_name(pricing_rule['backend_system_name'])
                end
metric = metric_owner.find_metric_or_method(pricing_rule['metric_system_name'])
                              
                             
```

Reply:
and the same code in for the limits

Reply:
:thinking: your alternative proposal looks good to me

Reply:
implemented